src/sema/types.rs

```diff
@@ -197,37 +204,117 @@ fn type_decl(
     });
 }
 
-/// check if a struct contains itself. This function calls itself recursively
-fn find_struct_recursion(struct_no: usize, structs_visited: &mut Vec<usize>, ns: &mut Namespace) {
-    let def = ns.structs[struct_no].clone();
-    let mut types_seen: HashSet<usize> = HashSet::new();
-
-    for (field_no, field) in def.fields.iter().enumerate() {
-        if let Type::Struct(StructType::UserDefined(field_struct_no)) = field.ty {
-            if types_seen.contains(&field_struct_no) {
-                continue;
+/// A struct field is considered to be of infinite size, if it contains itself infinite times.
+///
+/// This function sets the `infinitie_size` flag accordingly for all connection between `nodes`.
+/// `nodes` is assumed to be a set of strongly connected nodes from within the `graph`.
+///
+/// Any node (struct) can have one or more edges (types) to some other node (struct).
+/// A struct field is not of infinite size, if there are any 2 neighboring nodes in the path,
+/// where all connecting edges between any two nodes in the `path` are mappings or dynamic arrays.
```

Comment:
Path is the sequence of distinct edges between two vertices. The semantic of the word is not correct here, because you do not specify which vertices you are accounting for. The set of all edges in a graph is not necessarily a path.

In addition, you first say `there are any two neighboring nodes` and later you say `any two nodes`. It looks like the former reference is disconnected with the latter reference.

I suggest this sentence be reformulated.

Reply:
Ok, exchanged `path` with `connection`. Please keep in mind we are not trying to write a formal book about graph theory here. This is just a doc comment providing additional context

Reply:
I have no problem parsing/understanding the comment, it's perfectly clear.